packages/fluentui/react-builder/src/components/ReaderNarration.tsx

```diff
@@ -16,98 +17,119 @@ export type ReaderNarrationProps = {
 
 export const ReaderNarration: React.FunctionComponent<ReaderNarrationProps> = ({ selector, inUseMode }) => {
   const ref = React.useRef<HTMLElement>();
-  const [selectedNarrationPath, setSelectedNarrationPath] = React.useState(null);
+  const [narrationElement, setNarrationElement] = React.useState(null);
   const [narrationText, setNarrationText] = React.useState('');
 
-  // Computes and saves the narration paths and texts for the given parent element and its focusable descendants
-  const computeAndSave = React.useCallback(element => {
-    computer.compute(element, 'Win/JAWS').then(narrations => {
-      narrationPaths = [];
-      narrationTexts = {};
-      narrations.forEach(narration => {
-        // Begin forEach 1
-        const path = narration.path.join(' > ');
-        narrationPaths.push(path);
-        narrationTexts[path] = narration.text;
-      }); // End forEach 1
-
-      // If narration path has not been selected by user, preselect the first path and its associated narration as defaults
-      const text = narrationPath == null ? narrations[0]?.text || null : narrationTexts[narrationPath];
-      if (narrationPath == null && narrationPaths.length > 1) {
-        // Begin if 1
-        setSelectedNarrationPath(narrationPaths[0]);
-      } // End if 1
-
-      setCompleteText(text);
-    }); // End compute
-  }, []); // End computeAndSave
-
   // Sets the complete screen reader narration text to be displayed.
   const setCompleteText = text => {
     setNarrationText(text !== null ? `Narration: ${text}` : null);
   }; // End setCompleteText
 
-  // Handles the "focusin" event by computing and saving the narration paths and texts.
+  // Handles the element path dropdown change event by updating the current and previous narration elements.
+  const handleElementPathChange = (event: any, props: DropdownProps) => {
```

Comment:
```suggestion
  const handleElementPathChange = (event: Event, props: DropdownProps) => {
```

Reply:
If I apply your suggestion, I get: `TS2322: Type '(event: Event, props: DropdownProps) => void' is not assignable to type '(event: MouseEvent<Element, MouseEvent> | KeyboardEvent<Element>, data: DropdownProps) => void'.`

Reply:
shouldn't this be one of the events inherited from React.SyntheticEvent ? so MouseEvent if it's a click

Reply:
Neither MouseEvent works.

Reply:
However, `const handleElementPathChange = (event: MouseEvent | KeybordEvent, props: DropdownProps)` wowrks. Should I leave it like that?

Reply:
I am sorry, MouseEvent | KeyboardEvent does not work either.

Reply:
In that case as we discussed offline React.SyntheticEvent will work since it's the parent of all react events... not very clean but I guess better than any :)